gpt_index/vector_stores/pinecone.py

```diff
@@ -35,6 +35,19 @@ def get_node_info_from_metadata(
     return node_extra_info
 
 
+def batch_upserts(
```

Comment:
mostly a nitpick but let's use the `itertools` version, and also we can pull this out into a general utility file. 

e.g. 
```
def chunks(iterable, batch_size=100):
    """A helper function to break an iterable into chunks of size batch_size."""
    it = iter(iterable)
    chunk = tuple(itertools.islice(it, batch_size))
    while chunk:
        yield chunk
        chunk = tuple(itertools.islice(it, batch_size))
```

Reply:
sounds good, moved it to a utils.py file in the vector_stores folder.